firebase-installations/src/test/java/com/google/firebase/installations/FirebaseInstallationsTest.java

```diff
@@ -140,7 +141,10 @@ public class FirebaseInstallationsTest {
   public void setUp() {
     MockitoAnnotations.initMocks(this);
     FirebaseApp.clearInstancesForTest();
-    executor = new ThreadPoolExecutor(0, 1, 30L, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+    backgroundExecutor =
```

Comment:
Can you use `com.google.firebase.concurrent.TestOnlyExecutors` ?

Reply:
Tried to use a few of the executors available there, but they all seem to cause tests to fail from too many invocations. I spent a bit of time trying to figure out why, but ended up rolling back the change.

Reply:
Looking into why this is the case more careful, I think tests are failing correctly. Using the `TestOnlyExecutors` in the tests do cause the backend to be called multiple times because it's being called concurrently multiple times.

The source of the issue is that previously FIS was using a single-thread executor for all tasks, which meant that they were not actually being executed concurrently. 

https://github.com/firebase/firebase-android-sdk/blob/master/firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java#L93

Moving FIS to common executors is changing this, thus making the test to fail.

We could either see if making FIS truly concurrent is viable, or migrate them to use sequential executor

https://firebase.github.io/firebase-android-sdk/components/executors/#sequential-executor

Reply:
After looking through the synchronization model here it seems that the network executor is the problem

The concurrent network requests just shouldn't happen, so I've replaced the network executor with a sequential one, i've also replicated this setup in the test, and this appears to have fixed it.